geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -224,42 +224,46 @@ object DeltaWriter extends StrictLogging {
   }
 
   /**
-    * Reduce function for delta records created by DeltaWriter
-    *
-    * @param sft simple feature type
-    * @param dictionaryFields dictionary fields
-    * @param encoding simple feature encoding
-    * @param sort sort
-    * @param batchSize batch size
-    * @param deltas output from `DeltaWriter.encode`
-    * @return single arrow streaming file, with potentially multiple record batches
-    */
+   * Reduce function for delta records created by DeltaWriter
+   *
+   * @param sft simple feature type
+   * @param dictionaryFields dictionary fields
+   * @param encoding simple feature encoding
+   * @param sort sort metadata, if defined each delta is assumed to be sorted
+   * @param sorted whether features are already globally sorted or not
+   * @param batchSize batch size
+   * @param deltas output from `DeltaWriter.encode`
+   * @return single arrow streaming file, with potentially multiple record batches
+   */
   def reduce(
       sft: SimpleFeatureType,
       dictionaryFields: Seq[String],
       encoding: SimpleFeatureEncoding,
       sort: Option[(String, Boolean)],
+      sorted: Boolean,
       batchSize: Int,
       deltas: CloseableIterator[Array[Byte]]): CloseableIterator[Array[Byte]] = {
-    new ReducingIterator(sft, dictionaryFields, encoding, sort, batchSize, deltas)
+    new ReducingIterator(sft, dictionaryFields, encoding, sort, sorted, batchSize, deltas)
   }
 
   /**
-    * Merge without sorting
-    *
-    * @param sft simple feature type
-    * @param dictionaryFields dictionary fields
-    * @param encoding simple feature encoding
-    * @param mergedDictionaries merged dictionaries and batch mappings
-    * @param batchSize record batch size
-    * @param threadedBatches record batches, grouped by threading key
-    * @return
-    */
+   * Merge without sorting
+   *
+   * @param sft simple feature type
+   * @param dictionaryFields dictionary fields
+   * @param encoding simple feature encoding
+   * @param mergedDictionaries merged dictionaries and batch mappings
+   * @param sort sort metadata for file headers
+   * @param batchSize record batch size
+   * @param threadedBatches record batches, grouped by threading key
+   * @return
+   */
   private def reduceNoSort(
       sft: SimpleFeatureType,
       dictionaryFields: Seq[String],
       encoding: SimpleFeatureEncoding,
       mergedDictionaries: MergedDictionaries,
+      sort: Option[(String, Boolean)],
```

Comment:
This is indicating the field to sort and whether or not it is ascending?  

Makes me think of the GeoTools Sort/SortOrder classes.

Reply:
yeah it's equivalent to that - maybe we should extract it out into a case class or trait.

Reply:
Do you want to extract that as a class now or later?

Reply:
we can do that later...